tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -0,0 +1,355 @@
+
+  def update(self, state, new_sample, axis=None):
+    """Update the running statistics of the `RunningCovarianceState` with a
+    new sample using the covariance update formula proposed by Philippe
+    Pebay (2008).
+
+    This implementation supports both batched and chunked covariance
+    computation. A "batch" is the usual parallel computation, namely a batch
+    of size N implies N independent covariance computations, each stepping one
+    sample (or chunk) at a time. A "chunk" of size M implies incorporating M
+    samples into a single covariance computation at once, which is more
+    efficient than one by one.
+
+    Args:
+      state: `RunningCovarianceState` that depicts the current state of running
+        statistics
+      new_sample: Incoming sample with shape and dtype compatible with those
+        used to form the `RunningCovarianceState`
+      axis: If chunking is desired, this is an integer that specifies the axis
+        with chunked samples. For individual samples, set this to `None`. By
+        default, samples are not chunked (`axis` is None)
+
+    Returns:
+      state: `RunningCovarianceState` with updated calculations
+
+    References:
+    [1]: Philippe Pebay. Formulas for Robust, One-Pass Parallel Computation of
+    Covariances and Arbitrary-Order Statistical Moments. _Technical Report
+    SAND2008-6212_, 2008.
+    https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2008/086212.pdf
+    """
+    updated_state = _update_running_covariance(
+        state, new_sample, self.event_ndims, self.dtype, axis=axis)
+    return state._replace(**updated_state._asdict())
+
+  def finalize(self, state, ddof=0):
+    """Finalizes running covariance computation for the `state`
+
+    Args:
+      state: `RunningCovarianceState` that depicts the current state of running
+        statistics
+      ddof: Requested dynamic degrees of freedom for the covariance calculation.
+        For example, use `ddof=0` for population covariance and `ddof=1` for
+        sample covariance. Defaults to the population covariance.
+
+    Returns:
+      covariance: an estimate of the covariance
+    """
+    return _finalize_running_covariance(state, ddof=ddof)
+
+def _init_running_covariance(shape, event_ndims='full', dtype=tf.float32):
+  if event_ndims == 'full':
+    event_ndims = len(shape)
+  if event_ndims > len(shape):
+    raise ValueError('Cannot calculate cross-products in {} dimensions for '
+                     'samples of rank {}'.format(event_ndims, len(shape)))
+  if event_ndims > 13:
+    raise ValueError('`event_ndims` over 13 not supported')
+  extra_ndims_shape = () if event_ndims == 0 else shape[-event_ndims:]
+  return RunningCovarianceState(
+      num_samples=tf.zeros((), dtype=dtype),
+      mean=tf.zeros(shape, dtype=dtype),
+      comoment=tf.zeros(shape + extra_ndims_shape, dtype=dtype),
+    ), event_ndims
+
+
+def _generate_pairwise_deltas(delta_mean, event_ndims):
```

Comment:
A consideration I didn't make, thanks for letting me know! I'm assuming it would still be useful to lay out the current use-case in the Docstring, but please correct me if otherwise.

Reply:
I've convinced myself that describing a specific use-case may be more confusing that just leaving it general; hence, I now lean towards opting for the latter.